e2e/config/azure.go

```diff
@@ -208,7 +208,7 @@ func NewAzureClient(subscription string) (*AzureClient, error) {
 		return nil, fmt.Errorf("create vnet client: %w", err)
 	}
 
-	cloud.GalleryImageVersion, err = armcompute.NewGalleryImageVersionsClient(subscription, credential, opts)
+	cloud.GalleryImageVersion, err = armcompute.NewGalleryImageVersionsClient(gallerySubscription, credential, opts)
```

Comment:
I would just call `Config.GallerySubscriptionID` here. And do the same for subscription.